numpy/distutils/fcompiler/ibm.py

```diff
@@ -35,9 +38,15 @@ def get_version(self,*args,**kwds):
             lslpp = find_executable('lslpp')
             xlf = find_executable('xlf')
             if os.path.exists(xlf) and os.path.exists(lslpp):
-                s, o = exec_command(lslpp + ' -Lc xlfcmp')
-                m = re.search(r'xlfcmp:(?P<version>\d+([.]\d+)+)', o)
-                if m: version = m.group('version')
+                try:
+                    o = subprocess.check_output([lslpp] + ['-Lc', 'xlfcmp'])
+                except (OSError, subprocess.CalledProcessError):
+                    pass
+                else:
+                    o = filepath_from_subprocess_output(o)
```

Comment:
This function isn't appropriate - the result is not a filepath, but a version string.

`filepath_from_subprocess_output` exists to convert from stdout encoding to a best-guess of file-system encoding. You don't need the second half of that.

Reply:
Right, but `exec_command()` never made that distinction so applying the processing is simply matching the previous control flow.

I'll happily change it though--makes sense and seems more efficient & everything seems the same when I delete it locally.